File: python/bronze/25304.py
import sys

X = int(sys.stdin.readline())
N = int(sys.stdin.readline())
result = 0

for _ in range(N):
    temp1, temp2 = map(int, sys.stdin.readline().split())
    result += temp1 * temp2

if(result == X):
    print("Yes")
else:
    print("No")

File: C++/bronze/25304.cpp
#include <iostream>

using namespace std;

int main(){
    long X, result = 0;
    int N, a, b;
    cin >> X;
    cin >> N;
    for(int i=0; i<N; i++){
        cin >> a >> b;
        result += a*b;
    }
    if(result == X){
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }
    return 0;
}